pkg/server/service/tcp/service.go

```diff
@@ -91,10 +95,28 @@ func (m *Manager) BuildTCP(rootCtx context.Context, serviceName string) (tcp.Han
 				continue
 			}
 
+			tcpAddr, err := net.ResolveTCPAddr("tcp", server.Address)
+			if err != nil {
+				srvLogger.Error().Err(err).Msg("Failed to resolve TCP address")
+				continue
+			}
+			healthCheckTargets[server.Address] = tcpAddr
+
 			loadBalancer.AddServer(handler)
 			logger.Debug().Msg("Creating TCP server")
 		}
 
+		if conf.LoadBalancer.HealthCheck != nil {
+			m.healthCheckers[serviceName] = healthcheck.NewServiceTCPHealthChecker(
+				m.dialerManager,
+				nil,
+				conf.LoadBalancer.HealthCheck,
+				loadBalancer,
+				conf,
+				healthCheckTargets,
+				serviceQualifiedName)
+		}
+
 		return loadBalancer, nil
 
 	case conf.Weighted != nil:
```

Comment:
I noticed the health check mechanism is not implemented for WRR. Is it intended?

Reply:
Indeed, I was thinking WRR could come with a separate PR as there might be other things to consider with the healthcheck, but didn't actually look further on it.